lib/runtime-core/src/parse.rs

```diff
@@ -369,7 +368,16 @@ pub fn read_module<
 
                 info.write().unwrap().globals.push(global_init);
             }
-
```

Comment:
@losfair I think the issue here was modules without function bodies, so the following wasn't called. Please review.